cms/static/js/spec/views/move_xblock_spec.js

```diff
@@ -0,0 +1,345 @@
+                            {
+                                category: 'vertical',
+                                display_name: 'unit_display_name_0',
+                                id: 'unit_ID'
+                            },
+                            {
+                                category: 'sequential',
+                                display_name: 'subsection_display_name_0',
+                                id: 'subsection_ID'
+                            },
+                            {
+                                category: 'chapter',
+                                display_name: 'section_display_name_0',
+                                id: 'section_ID'
+                            },
+                            {
+                                category: 'course',
+                                display_name: 'Demo Course',
+                                id: 'COURSE_ID_101'
+                            }
+                        ]
+                    };
+
+                renderViews(outline, ancestorInfo);
+                verifyXBlockInfo(outlineOptions, 'section', 0, 'forward', true);
+                // click the outline breadcrumb to render sections
+                moveXBlockBreadcrumbView.$el.find('.bc-container button').first().click();
+                verifyXBlockInfo(outlineOptions, 'section', 1, 'forward', false);
+            });
+
+            it('shows correct message when parent has no children', function() {
+                var outlinesInfo = [
+                    {
+                        outline: createCourseOutline({}),
+                        message: 'This course has no sections'
+                    },
+                    {
+                        outline: createCourseOutline({section: 1}),
+                        message: 'This section has no subsections',
+                        forwardClicks: 1
+                    },
+                    {
+                        outline: createCourseOutline({section: 1, subsection: 1}),
+                        message: 'This subsection has no units',
+                        forwardClicks: 2
+                    },
+                    {
+                        outline: createCourseOutline({section: 1, subsection: 1, unit: 1}),
+                        message: 'This unit has no components',
+                        forwardClicks: 3
+                    }
+                ];
+
+                _.each(outlinesInfo, function(info) {
+                    renderViews(info.outline);
+                    _.each(_.range(info.forwardClicks), function() {
+                        clickForwardButton();
+                    });
+                    expect(moveXBlockListView.$el.find('.xblock-no-child-message').text().trim()).toEqual(info.message);
+                    moveXBlockListView.undelegateEvents();
+                    moveXBlockBreadcrumbView.undelegateEvents();
+                });
+            });
+        });
+    });
```

Comment:
@andy-armstrong There is second following [PR ](https://github.com/edx/edx-platform/pull/14352) which includes enable/disable button along with move & undo functionality.